### PR TITLE
test(e2e): migrate Gateway API tests to e2e_env

### PR DESCRIPTION
**Pull request by michaelbeaumont**
This doesn't include the conformance tests.

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? -- https://github.com/kumahq/kuma/issues/4520
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](/CONTRIBUTING.md#backporting)? --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/e2e_env/kubernetes/gateway/gatewayapi.go [hunk 1]
```diff
@@ -31,79 +25,71 @@ spec:
...
 	})
-})
 
-func GatewayAPI() {
-	if Config.IPV6 {
-		return // KIND which is used for IPV6 tests does not support load balancer that is used in this test.
-	}
+	E2EAfterAll(func() {
+		Expect(env.Cluster.DeleteNamespace(namespace)).To(Succeed())
+		Expect(env.Cluster.DeleteNamespace(externalServicesNamespace)).To(Succeed())
```
Comment (jakubdyszkiewicz) [inline test/e2e_env/kubernetes/gateway/gatewayapi.go:83]: Change this to `TriggerDeleteNamespace`. We don't want to wait for the namespace to be deleted.

**Review by jakubdyszkiewicz — COMMENTED** [review]
nice!

**Comment by jakubdyszkiewicz** [comment]
The legacy test slice with `./test/e2e/gateway/... ./test/e2e/virtualoutbound/...` now takes 16min :) It was ~50min before we started migrating any gateway tests

File: test/e2e_env/kubernetes/gateway/gatewayapi.go (hunk 1 above)
Comment (michaelbeaumont) [reply test/e2e_env/kubernetes/gateway/gatewayapi.go:83]: Indeed, missed that

**Review by jakubdyszkiewicz — APPROVED** [review]

**Comment by michaelbeaumont** [comment]
After changing the port, the [e2e tests haven't failed (last 2 pipelines)](https://app.circleci.com/pipelines/github/kumahq/kuma?branch=pull%2F4796). Merging

**Pull request merged**